### DISCUSSION TITLE
documentation on custom router

### DISCUSSION THREAD

**Discussion by bluebrown** [go-swagger/go-swagger#2688] [Q&A] [answered]
Hi, in the docs, it's mentioned that we can use a different router.  Is there some documentation on how to set this up?

> The default router for go-swagger is naoina's denco which is a very fast ternary search tree based router that allows for much greater flexibility than the trie based router implementation of julienschmidt at almost the same and sometimes lower cost.
https://goswagger.io/generate/server.html

I think the denco router has not much adoption found in its 8+ years existence. I would like to use chi for example, which is more widely used.

**Comment by casualjim** [comment, marked as answer]
it's a hidden component, and it composes with other routers. Everything is behind a net/http.Handler anyway so you it's not like it's infecting your code base.

You need to implement the router interface: https://github.com/go-openapi/runtime/blob/master/middleware/router.go#L106-L110
